lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -534,10 +616,98 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
             StringRef(buf, numBytes));
       }
 
-      // Close the read end of the pipe.
+      if (readfds[0].revents & POLLIN) {
+        ssize_t numBytes = read(releasePipe[0], buf, sizeof(buf));
+        if (numBytes < 0) {
+          int err = errno;
+          getDelegate().commandProcessHadError(
+               context.job.getForCommand(), handle,
+               Twine("unable to read process release signal (") + strerror(err) + ")");
+          break;
+        }
+
+        if (numBytes == 0) {
+          readfds[0].events = NULL;
+          if (shouldCaptureOutput) {
+            continue;
+          }
+          break;
+        }
+
+        auto msg = StringRef(buf, numBytes).trim();
```

Comment:
I would prefer we not trim, but check for a very precise & strict protocol.

Reply:
The trim was to allow for easy of use with things such as "echo ${LLBUILD_TASK_ID} >&${LLBUILD_LANE_RELEASE_FD}".   I could explicitly check to newline terminated values, though.

Reply:
In the spirit of generality (re: SR-6053), should this also be something like "release ${LLBUILD_TASK_ID}\n"?

Reply:
I suspect we will want to change the binary protocol, so I'd probably recommend something like
`v1\n${LLBUILD_TASK_ID}\n`. Maybe `llbuild.v1` if we want it to be less ambiguous. I don't think the overhead of the extra bytes will matter, but it will be easier to rev in the future.

Also, you can use printf(1) for more precise output in a shell script.